### DISCUSSION TITLE
Using EMS-ESP32 to "call for heat" or in some other way simulate heat demand

### DISCUSSION THREAD

**Discussion by srchild** [emsesp/EMS-ESP32#1613] [General]
- I do not have easy access to the LR and LS terminals on my boiler, which is installed outside with the through-the-wall pipes and cables neatly boxed off with no way for me to thread extra cables. So far as I can see LR and LS are not brought indoors.

 - I do have easy access to the EMS bus, via the receiver section of the Comfort RF2 wireless thermostat/timer which is installed indoors (and easily accessible), hence asking about EMS-ESP32.

My boiler is a 32kW external Worcester Greenstar Heatslave oil boiler (installed 2018). I want to be able to signal "call to heat" (heat demand) to my boiler. Can I do this (or achieve the same effect) using EMS-ESP32?

From reading these pages I understand that the EMS does not offer a "call to heat" function as such, but a similar effect can be obtained by manipulating the flow temperature? I also read that an EMS connected thermostat will override such manipulation. Probably I would disconnect the Comfort RF2, and provide timer and heating settings via Home Assistant alone (see below). For us the Comfort RF2 thermostat function is the problem (see below), and the device is not of any use to us other than as a timer.

The layout here is complex, but in effect the single boiler is heating three adjacent properties with no zoning set up and no provision for anything other than the single thermostat (which is in the property nearest to the boiler) to "call for heat".

The area furthest away from the boiler has an underfloor heating installation supplied with heat from a thermal store. However the principal source of heat for the thermal store is a loop from the Worcester boiler radiator piping. (It works!)

The thermal store has no way of triggering the boiler when it needs heat. The thermostat (2 properties away) has to be set to a high temperature to ensure that the boiler is on. Anyway we already run the thermostat set permanently to a high temperature, even when not using the thermal store, to enable heating of the other two properties (which have zigbee TRV on all radiators).

In effect the thermostat is already disabled. Of course this means the boiler is cycling whenever heat is not needed.

My plan is 

 - set up a bbqkees gateway, connected to the EMS bus at the Comfort RF2 receiver location (perhaps with a switch to flip the bus between Comfort RF2 amd gateway, to provide a fallback in the event of problems with home assistant or gateway)

 - configure home assistant automation to run the heating and DHW scheduling

 - configure home assistant automation to call for heat when heating scheduling is on and the thermal store is requiring heat (as determined by a home assistant automation which is already operating to open a motorised valve on the radiator loop connected to the thermal store)

 - configure home assistant to call for heat when heating scheduling is on and any TRV anywhere has local_temperature < current_heating_setpoint (for at least say 5-10 minutes to reduce cycling)

 - otherwise don't call for heat (set flow temperature to 0?)

My home assistant is somewhat "hardened" to make it suitable for such a core function. e.g. I run it from SSD, not from SDcard. I have a fallback spare RPi (with A2 SDcard) ready to swap in and restore from backup in the event of failure of the primary RPi/SSD. And as per above I could retain the possibility to switch back to the Comfort RF2 if required (e.g. when the service engineer visits for the annual service!)

Will it work? Is it sensible? Is there a better way? Your views are welcome.

**Comment by srchild** [comment]
Any comments on this? Specifically, am I correct in thinking that I can reduce boiler cycling by manipulating flow temperature (setting it low when there is no heat needed, and restoring it to a higher value when heat is needed)?

Thanks

**Reply by proddy** [reply]
it's beyond me, @bbqkees any input?

**Comment by srchild** [comment]
Perhaps I could broaden the question.

If I install a bbqkees gateway, are there values exposed that I could manipulate to achieve a similar benefit to signalling heat demand in the conventional way:

 - to ensure that the boiler is generating heat whenever a thermostat or other devices requires it
 - to avoid generating heat when no device is calling for it, i.e. reduce boiler cycling

Thanks

**Comment by bergteam** [comment]
This issue is faced in lot's of discussions "How to increase / decrease Heating", also by myself.

My -plain- Setup: Junkers ZSB-14-4C with FW120 Thermostat directly mounted on Boiler, Set to weather controled.

Understood so far:
+ Thermostat input outside temperature and room temperatur settings, calculates "hc1 selected room temperature" and sends this to Boiler
+ Boiler calculates Pump and Burner Power => Fully Controls Burner
+ Heating Increase / Decrease should be set on Thermostat
+ There are issues as the Thermostat does not expose those settings

Ideas:
1. Currently I control heating from outside (iobroker) by temporarily setting the "hc1 heat/eco/nofrost temperature" values
2. Whishlist: Temporarily change the Thermostat heat/eco/nofrost mode, and on next switching point, the Thermostat returns to it's program
=> Why is this not possible? There is a "Thermostat Control Device, called FB10" which can set exactly this from outside to the Thermostat via the Bus.
=> EMS-ESP should be able to set it to. How can I help development in finding this out?

**Reply by proddy** [reply]
we need someone with a FB10 to sniff the telegrams so EMS-ESP can simulate it. You could scout the Github issues and see who can do this for us. https://emsesp.github.io/docs/Troubleshooting/#im-missing-certain-data-from-an-ems-device

**Reply by MichaelDvP** [reply]
For FB10 emulation see #1602

**Reply by MichaelDvP** [reply]
I've found #728 about the same problem, but in the tests were a mistake sending to boiler instead of master-thermostat. 
@bergteam Can you try with a writeable custom entity device 0x10, type 0x123, offset 2, Valuetype int, factor 0.5

**Reply by bergteam** [reply]
I've read those posts and am not sure what was tried:
+ They have a physical FB10 Remote and see this device in EMS-ESP

My Setup is:
+ FW120 without FB10, I just want to control heat on/off somewhow from outside

Another Idea:
+ manually set "hc1 mode" to "nofrost" or "eco" and then back to "auto"
+ somehow I have not seen the "auto" mode in previous EMS-ESP Versions
=> That would fit my use case perfectly.

+ once I switched to "heat" or "nofrost" previously from EMS-ESP then "Netcom Activated" appears on the FW120, which I assume is fine

=> I can test after easter holidays. No experiements before :-)

**Reply by bergteam** [reply]
Test was successful:
+ Boiler Mode = auto
=> On EMS-ESP set Mode = eco  //Netcom appears on FW120 display, mode set to eco
+ Meanwhile time reached where FW120 would switch to nofrost. Nothing happens, still in eco mode = OK.
=> On EMS-ESP set Mode = auto  //Netcom remains on FW120 display, immediately nofrost mode choosen = OK

This fully covers my use case, to control the system from outside and give control back to FW120. Thanks to your great piece of software!

One small question remains: Why does "Netcom" appear on FW120 display? This never goes away, even if switched back to auto by EMS-ESP. This may confuse others, trying to operate the boiler on the FW120.
